drivers/serial/serialcore.c

```diff
@@ -21,15 +21,15 @@ int serial_gets(struct serial_info *serial,
     return serial->ops->serial_gets(serial, len, retlen, buf);
 }
 
-int serial_putc(struct serial_info *serial, char c)
+void serial_putc(struct serial_info *serial, char c)
```

Comment:
Why don't we just follow API behavior as C standard library? That is, `putc` returns the character written as an unsigned char cast to an int or EOF on error, and `puts()` returns a nonnegative number on success, or EOF on error. The parameter list can be revised as well.

Reply:
That might be better at this stage. I'll try to import `stdio.h` which defines EOF as -1.